subsys/net/ip/tcp2.c

```diff
@@ -916,10 +929,13 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 	case TCP_CLOSED:
 		tcp_conn_unref(conn);
 		break;
-	case TCP_TIME_WAIT:
 	case TCP_CLOSING:
 	case TCP_FIN_WAIT_1:
 	case TCP_FIN_WAIT_2:
+	case TCP_TIME_WAIT:
+		k_delayed_work_submit(&conn->timewait_timer,
+				      CONFIG_NET_TCP_TIME_WAIT_DELAY);
```

Comment:
Because of k_timeout_t changes, the timeout needs to be written like this:
`K_MSEC(CONFIG_NET_TCP_TIME_WAIT_DELAY)`

Reply:
@jukkar 

OK, thanks. Fixed.